internal/castai/castai.go

```diff
@@ -86,26 +88,34 @@ func (c *client) SendDelta(ctx context.Context, clusterID string, delta *Delta)
 		return fmt.Errorf("invalid url: %w", err)
 	}
 
-	r, w := io.Pipe()
+	b, err := json.Marshal(delta)
```

Comment:
Using json encoder as it was before could be a bit more memory efficient as it doesn't need to hold whole json temporary in memory.